#!/usr/bin/env python
PACKAGE = "rr_common"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Planner Params
# Must be 10 or less provided the matrix size
gen.add("n_segments", int_t, 0, "planner: segment size", 7, 1, 10)

# Bicycle Model Params
gen.add("max_lateral_accel", double_t, 0, "bicycle model: max lateral acceleration", 7.0, 0.0, 14.0)
gen.add("segment_size", int_t, 0, "bicycle model: segment size", 25, 1, 50)
gen.add("dt", double_t, 0, "bicycle model: dt", 0.02, 0.001, 2.0)

# Control Constants
gen.add("steering_gain", double_t, 0, "steering gain", 1.4, 0.00, 3.0)
gen.add("k_map_cost", double_t, 0, "cost: map constant", 0.1, 0.00, 1.0)
gen.add("k_global_path_cost", double_t, 0, "cost: global map cost", 0.0, -10.0, 100)
gen.add("k_speed", double_t, 0, "cost: speed constant", 0.05, 0.00, 1.0)
gen.add("k_steering", double_t, 0, "cost: steering constant", 0.01, 0.00, 1.0)
gen.add("k_angle", double_t, 0, "cost: angle constant", 0.00, 0.00, 1.0)
gen.add("collision_penalty", double_t, 0, "cost: collision penalty", 10000, 100, 20000)

exit(gen.generate(PACKAGE, "rr_common", "planner_configuration"))